wandb/sdk/wandb_artifacts.py

```diff
@@ -76,6 +78,14 @@ def __init__(self, entry: ArtifactEntry, obj: data_types.WBValue):
         self.obj = obj
 
 
+def _normalize_metadata(metadata: Optional[Dict[str, Any]]) -> Dict[str, Any]:
+    if metadata is None:
+        return {}
+    if not isinstance(metadata, dict):
+        raise TypeError(f"metadata must be dict, not {type(metadata)}")
+    return cast(Dict[str, Any], json.loads(json.dumps(metadata)))
```

Comment:
😍 I learn something every time I review your PRs!! This is a very nice trick.
I wonder what it does to mypy's idea of coverage, i.e. I usually do:
```python
result: Dict[str, Any] = json.loads(json.dumps(metadata))
return result
```
but that only partially satisfies mypy (yellow)

Reply:
Hmm! Interesting, I would've thought Mypy would accept your approach too. I guess `cast` is more explicitly saying "Mypy, I know better than you here, just take my word for it," so Mypy is more deferent to it.